### PR TITLE
APIS-2816 - Adding a String to Int operation to jstransform

### PR DESCRIPTION
**Pull request by adilbozai1995**

### CODE REVIEW

**Comment by paas-api-pr-manager** [comment]
Jira Tickets included in PR are:

* [APIS-2816](https://jira.gannett.com/browse/APIS-2816): Add a String to Int operation to jstransform 
```
In some cases we may want to cast the result of the of the transformation to integer.
 For example, we may need to perform a _replace_ operation on *id*, and then cast the result of _replace_ to integer (_replace_ returns a string)
{noformat}
"id": {
  "type": "string",
  "transform": {
    "sport": {
      "from": [
        {
          "xmlPath": "id",
          "operations": [
            {
              "type": "replace",
              "args": {
                "regex": "[\\D]",
                "new": ""
              }
            }
          ]
        }
      ]
    }
  }
}
{noformat}
For that we may delevlop a new jstransform operation, so the trasformation will look like
{noformat}
"operations": [
    {
      "type": "replace",
      "args": {
        "regex": "[\\D]",
        "new": ""
      }
    },
    {
      "type": "atoi"
    }
  ]
{noformat}
*AC*
 develop a new _jstransform_ operation that accepts a string, tries to cast it to an integer and returns this integer
```

File: transform/operations.go [hunk 1]
```diff
@@ -246,6 +246,28 @@ func (t *timeParse) transform(raw interface{}) (interface{}, error) {
...
+
+func (s *stringToInteger) transform(raw interface{}) (interface{}, error) {
+	str, ok := raw.(string)
+	if !ok {
+		return nil, errors.New("stringToInteger only supports strings")
+	}
+
+	result, err := strconv.Atoi(str)
+	if err != nil {
+		return nil, errors.New(fmt.Sprintf("error converting to an Integer from a string: %s", err))
```
Comment (abaehre) [inline transform/operations.go]: if you're going to use fmt and errors you can probably just use fmt.Errorf()

Comment (abaehre) [reply transform/operations.go]: actually you probably don't need most of this

you could just 
`return strconv.Atoi(str)`

File: transform/operations_test.go [hunk 2]
```diff
@@ -374,3 +374,28 @@ func TestTimeParse(t *testing.T) {
...
+	tests := []opTests{
+
+		{
+			description: "Simple working case",
+			in: "237754",
+			want: 237754,
+		},
+
+		{
+			description: "An alphabet input",
```
Comment (abaehre) [inline transform/operations_test.go]: alpha*

File: transform/operations_test.go [hunk 3]
```diff
@@ -374,3 +374,28 @@ func TestTimeParse(t *testing.T) {
...
+			in: "a",
+			wantErr: true,
+		},
+
+		{
```
Comment (abaehre) [inline transform/operations_test.go]: don't really need this case.

**Pull request merged**